src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs

```diff
@@ -60,20 +62,42 @@ public class PackageSource : IEquatable<PackageSource>
         /// </summary>
         public int ProtocolVersion { get; set; } = DefaultProtocolVersion;
 
+        /// <summary>
+        /// Whether the source is using the HTTP protocol, including HTTPS.
+        /// </summary>
         public bool IsHttp
         {
             get
             {
                 if (!_isHttp.HasValue)
                 {
-                    _isHttp = Source.StartsWith("http://", StringComparison.OrdinalIgnoreCase) ||
-                              Source.StartsWith("https://", StringComparison.OrdinalIgnoreCase);
+                    _isHttp = IsHttps || Source.StartsWith("http://", StringComparison.OrdinalIgnoreCase);
```

Comment:
Just another thought should be switch the position of `IsHttps` to last? If it's just "http" then it would short circuit. But same time https is more prevalent, then current one is ok.

Reply:
I moved it first because if we are going to check it either way, checking it first will possibly find "https", meaning we don't have to search "http".

Reply:
We could do more refactoring in the future. For example, just look for `StartsWith("http")` ? Maybe do URI parsing earlier so we already know the protocol used?